spotlight-client/src/UiLibrary/Dropdown/Dropdown.tsx

```diff
@@ -94,23 +98,74 @@ const DropdownMenuItem = styled.li`
 `;
 
 const DropdownMenuItemContents = styled(animated.div)`
+  display: flex;
+  justify-content: space-between;
   padding: ${rem(8)} ${rem(16)};
 `;
 
+const CheckMarkWrapper = styled.span`
+  display: inline-block;
+  margin-left: ${rem(32)};
+`;
+
 type DropdownOption = {
   id: string;
   label: string;
   hidden?: boolean;
+  color?: string;
 };
 
-const Dropdown: React.FC<{
+type SingleSelectProps = {
   disabled?: boolean;
   label: string;
   onChange: (id: string) => void;
+  onChangeMultiple?: never;
+  onHighlight?: (id?: string) => void;
   options: DropdownOption[];
   selectedId: string;
-}> = ({ disabled, selectedId, label, onChange, options: allOptions }) => {
-  const visibleOptions = allOptions.filter(({ hidden }) => !hidden);
+  multiple?: never;
+};
+
+type MultipleSelectProps = Assign<
+  SingleSelectProps,
+  {
+    multiple: true;
+    onChange?: never;
+    onChangeMultiple: (ids: string[]) => void;
+    selectedId: string[];
+  }
+>;
+
+function Dropdown(props: SingleSelectProps): React.ReactElement | null;
```

Comment:
There is quite a bit of logic in this component now checking whether it is a single select or multiple select. I wonder if it would be more clean to have two different components, a single select dropdown and a multi-select dropdown?  I don't feel strongly about it, but it might be worth considering.

Reply:
hmmm ... maybe? I thought about this but I'll admit I didn't actually try it. in retrospect I agree that there is an awful lot of branching going on here that suggests the component is trying to do too many things. I will take a look!

Reply:
Curious what you think of this change before merging ... because the two have so much in common (styling, most of the animations) I had a choice between doing a bunch of boilerplate composition or encapsulating the common parts and passing some obscure-ish props to cover the parts that diverge (namely: the button contents, some of the Downshift hook arguments, and how to render the options). I chose the latter!

Reply:
So sorry this has taken so long to get back to! Lost track of it with the long weekend. 

I like this implementation more than the single `Dropdown` with lots of branching logic. It is easier to read and not have to skim through large chunks of code when I'm looking at it through the 'multi' or 'single' lense. Also agree that the latter was a good approach. We do something similar in pulse to adapt a chart container to support multiple charts and I have found it pretty nice to work with.